### DISCUSSION TITLE
How to detect a solution is open

### DISCUSSION THREAD

**Discussion by GFlisch** [VsixCommunity/Community.VisualStudio.Toolkit#345] [Q&A]
Hi,

I would like to detect when a solution is opened to activate or not some features.

In the Package class I do this.

protected override async Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
        {
            this.AddService(typeof(IOutputWnd), CreateOutputWindowAsync, true);
            this.AddService(typeof(IContentService), CreateContentServiceAsync, true);
            
            await this.RegisterCommandsAsync();

            VS.Events.SolutionEvents.OnAfterOpenSolution += SolutionEvents_OnAfterOpenSolution;

        }

        private void SolutionEvents_OnAfterOpenSolution(Solution obj)
        {
            VS.MessageBox.Show("Welcome");
        }

But the event doesn't seem to be fired.

Any help how to achieve this?

Thanks.

**Comment by RobertvanderHulst** [comment]
Most likely your package is loaded after the solution has been loaded.
I suspect that it you close the solution and reopen it then you will see the event.
You can control when your package is loaded with the ProvideAutoLoad attribute, but still VS decides when you will be loaded.
Mads has a video about this on Youtube IIRC

**Comment by GFlisch** [comment]
@RobertvanderHulst ,

Ok. Thanks I will try to find the video.

**Comment by reduckted** [comment]
Using the event is the correct way to detect when a solution opens after your extension has loaded. Here's an example of how you can check if a solution is _already_ open when you're extension loads.

https://github.com/reduckted/ProjectFilter/blob/21d9754495e5282a5e300b0d766181a9172a4ea1/source/ProjectFilter/Services/SolutionLoadObserver.cs#L17-L21

**Reply by GFlisch** [reply]
Hi @reduckted.

Perfect 👌.. It works like a charm.

Thanks
